src/language-markdown/printer-markdown.js

```diff
@@ -752,13 +752,25 @@ function shouldPrePrintDoubleHardline({ node, previous, parent }, options) {
     parent.type === "listItem" &&
     previous.type === "paragraph" &&
     previous.position.end.line + 1 === node.position.start.line;
+  const isPrevNodeSpecificComment = ((node) => {
+    if (
+      node.type === "html" &&
+      node.value.startsWith("<!--") &&
+      node.value.endsWith("-->")
+    ) {
+      const SpecificCommentString = ["markdownlint-disable-next-line"];
+      return SpecificCommentString.some((s) => node.value.includes(s));
+    }
+    return false;
+  })(previous);
```

Comment:
I prefer to define the following function in `language-markdown/utils.js`:

```js
function isMarkdownlintComment(node) {
  return (
    node.type === "html" &&
    node.value.startsWith("<!--") &&
    node.value("-->") &&
    node.value === "markdownlint-disable-next-line"
  );
}
```

I also prefer variable name `isPrevNodeMarkdownlintComment`.